test/bench/portfft/launch_bench.hpp

```diff
@@ -135,10 +135,12 @@ void bench_dft_average_host_time_impl(benchmark::State& state, sycl::queue q, po
     }
     double elapsed_seconds =
         std::chrono::duration_cast<std::chrono::duration<double>>(end - start).count() / static_cast<double>(runs);
-    state.counters["flops"] = ops / elapsed_seconds;
-    state.counters["throughput"] = static_cast<double>(bytes_transferred) / elapsed_seconds;
+    state.counters["flops"] += ops / elapsed_seconds;
+    state.counters["throughput"] += static_cast<double>(bytes_transferred) / elapsed_seconds;
     state.SetIterationTime(elapsed_seconds);
   }
+  state.counters["flops"] /= static_cast<double>(state.iterations());
+  state.counters["throughput"] /= static_cast<double>(state.iterations());
```

Comment:
I think setting the counter every iteration works but isn't the intended way of doing this.

As far as I can see, we should have our own variable `flop_sum`, and then
```c++
state.counters["flops"] = Counter(flop_sum, benchmark::Counter::kAvgIterations);
```
See https://github.com/google/benchmark/blob/main/docs/user_guide.md#custom-counters and https://github.com/google/benchmark/blob/a008bf82f4aa5ad6605622040c7b2fe7512dc0c7/include/benchmark/benchmark.h#L654

(And if we don't do this, I went down this rabbit-hole whilst wondering about how the counters are initialized.)

Reply:
Done

Reply:
GBench already does the average from what I understand. Can you test this to make sure?

Reply:
GBench already does the average

I do not think so, the counter, is essentially a map from "key(flop / bandwidth)" to a double value;
What we were essentially doing was updating the map value, rather than taking the average.

>  Can you test this to make sure?

I would not how to test this with numbers to be honest, for a device with stable clock speeds etc, there would be no discernible difference between average and the last value, other than some which can be considered as benchmarking noise.

But from what I understand, portBLAS does something similar too

Reply:
portBLAS does not use `benchmark::Counter` from what I can see. From the comments in https://github.com/google/benchmark/blob/main/docs/user_guide.md#custom-counters this already does the division by the number of iterations. I'm not sure it does the summation though.
Sorry I didn't mean to test a difference between the average and the last result but to test that the division by the number of iteration is not done twice.

Reply:
> portBLAS does not use benchmark::Counter

No It does, but it does manually divide. 
Yes the Counter(..., kAvgIteration) divides it, and from what I understand, it simply divides the value passed to it by the number of iterations. I am  not manually dividing it either

Reply:
Sorry I looked at the diff too fast then, that looks good, thanks!